ci/scripts/run-codebuild/libs/codebuild-runner.test.js

```diff
@@ -0,0 +1,250 @@
+        deepLink: 'mockBuildLogDeepLink'
+      }
+    })
+
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 1 event 1')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 1 event 2')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 2 event 1')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 2 event 2')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 3 event 1')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 3 event 2')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 4 event 1')
+    expect(process.stdout.write).toHaveBeenCalledWith('mock message 4 event 2')
+  })
+
+  it('Allows the unhandled reject to bubble up after some logs are returned if codebuild fails during the build', async () => {
+    // Override the cwlMock response for an early request so the logs end early
+    cwlMock.on(GetLogEventsCommand, {
+      logGroupName: 'mockBuildLogGroupName',
+      logStreamName: 'mockBuildLogStreamName',
+      startFromHead: true,
+      nextToken: 'mockForwardToken1'
+    }).resolves({
+      events: [
+        { message: 'mock message 2 event 1' },
+        { message: 'mock message 2 event 2' }
+      ],
+      nextForwardToken: 'mockForwardToken1'
+    })
+
+    let batchGetBuildsCallCount = 0
+    cbMock
+      .on(StartBuildCommand, {
+        projectName: 'mockProject',
+        sourceVersion: 'mockSourceVersion',
+        secondarySourcesVersionOverride: [
+          { sourceIdentifier: 'mockSecondarySource1', sourceVersion: 'mockSecondarySource1Version' },
+          { sourceIdentifier: 'mockSecondarySource2', sourceVersion: 'mockSecondarySource2Version' }
+        ],
+        environmentVariablesOverride: [
+          { name: 'MOCK_ENV_VAR_1', value: 'MOCK_ENV_VAR_1_VALUE', type: 'PLAINTEXT' },
+          { name: 'MOCK_ENV_VAR_2', value: 'MOCK_ENV_VAR_2_VALUE', type: 'PLAINTEXT' }
+        ]
+      }).resolves(mockResponses.startBuildCommandResponse)
+      .on(BatchGetBuildsCommand, { ids: ['mockBuildId'] }).callsFake(input => {
+        if (batchGetBuildsCallCount === 0) {
+          batchGetBuildsCallCount++
+          return mockResponses.batchGetBuildsCommandResponse
+        } else {
+          throw Error('mocked rejection')
+        }
+      })
+
+    const runner = new CodeBuildRunner(
+      'mockProject',
+      'mockSourceVersion',
+      [
+        { sourceIdentifier: 'mockSecondarySource1', sourceVersion: 'mockSecondarySource1Version' },
+        { sourceIdentifier: 'mockSecondarySource2', sourceVersion: 'mockSecondarySource2Version' }
+      ],
+      [
+        { name: 'MOCK_ENV_VAR_1', value: 'MOCK_ENV_VAR_1_VALUE', type: 'PLAINTEXT' },
+        { name: 'MOCK_ENV_VAR_2', value: 'MOCK_ENV_VAR_2_VALUE', type: 'PLAINTEXT' }
+      ],
+      'eu-west-1',
+      0
+    )
+
+    await expect(runner.runBuildAndLog()).rejects.toThrow('mocked rejection')
```

Comment:
Is it worth asserting the `process.stdout.write` calls for these error cases like we do for the success case? ('No' is an acceptable answer here!)

Reply:
Added :) Thanks! Great spot!